### PR TITLE
Update Status field if a plugin gives an error

### PR DESCRIPTION
**Pull request by danielpygo**
- https://github.com/cloud-ark/kubeplus-operators/issues/34
 - Also, added GetSupportedPlugins to util
 - looping over only supported plugins in handlePod

### CODE REVIEW

File: moodle/moodlecontroller.go [hunk 1]
```diff
@@ -354,8 +354,11 @@ func (c *MoodleController) syncHandler(key string) error {
 
 			supportedPlugins = foo.Status.InstalledPlugins
 			supportedPlugins = append(supportedPlugins, installedPlugins...)
-
-			c.updateMoodleStatus(foo, podName, "", status, url, &supportedPlugins, &unsupportedPlugins)
+			if len(erredPlugins) > 0 {
+				c.updateMoodleStatus(foo, podName, "", "Error Plugins Not Ensured", url, &supportedPlugins, &unsupportedPlugins)
```
Comment (devdattakulkarni) [inline moodle/moodlecontroller.go]: "Error in installing some plugins"

File: moodle/moodle.go [hunk 2]
```diff
@@ -43,33 +43,19 @@ func (c *MoodleController) deployMoodle(foo *operatorv1.Moodle) (string, string,
...
-	for _, p := range plugins {
-		if _, ok := constants.PLUGIN_MAP[p]; ok {
-			supportedPlugins = append(supportedPlugins, p)
-		} else {
-			unsupportedPlugins = append(unsupportedPlugins, p)
-		}
-	}
-	fmt.Printf("MoodleController.go  : Supported Plugins: %v\n", supportedPlugins)
-	return supportedPlugins, unsupportedPlugins
+	return serviceURIToReturn, moodlePodName, secretName, unsupportedPlugins, erredPlugins, err
```
Comment (devdattakulkarni) [inline moodle/moodle.go]: Ok, so returning an err, which will be set in line 52. This is good.

File: moodle/moodle.go [hunk 3]
```diff
@@ -43,33 +43,19 @@ func (c *MoodleController) deployMoodle(foo *operatorv1.Moodle) (string, string,
...
 	plugins := foo.Spec.Plugins
 
-	supportedPlugins, unsupportedPlugins = c.getSupportedPlugins(plugins)
-
+	supportedPlugins, unsupportedPlugins = c.util.GetSupportedPlugins(plugins)
 	if len(supportedPlugins) > 0 {
 		namespace := getNamespace(foo)
-		fmt.Println("!!!!" + moodlePodName)
-		erredPlugins = c.util.EnsurePluginsInstalled(foo, moodlePodName, namespace, constants.PLUGIN_MAP)
+		erredPlugins = c.util.EnsurePluginsInstalled(foo, supportedPlugins, moodlePodName, namespace, constants.PLUGIN_MAP)
```
Comment (devdattakulkarni) [inline moodle/moodle.go]: Taking supportedPlugins as a parameter. This is good.

File: moodle/pkg/utils/utils.go [hunk 4]
```diff
@@ -133,12 +134,17 @@ func (u *Utils) EnsurePluginsInstalled(moodle *operatorv1.Moodle,
...
-	plugins := moodle.Spec.Plugins
 	erredPlugins := make([]string, 0)
-	for _, pluginName := range plugins {
+
+	for _, pluginName := range supportedPlugins {
 		fmt.Printf("  Installing plugin %s\n", pluginName)
-		pluginInstallDetails := pluginData[pluginName]
+		pluginInstallDetails, ok := pluginData[pluginName]
 
+		if !ok {
```
Comment (devdattakulkarni) [inline moodle/pkg/utils/utils.go]: Yeah, as we discussed this is probably dead code as we are already looping over supportedPlugins.

We can remove this.

**Comment by danielpygo** [comment]
Ok looks ready to me .

**Pull request merged**